File: Data Structures & Algorithms/search-2d-matrix/submission-4.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        def binarySearchRow(curArray : List[int]) -> bool:
            left = 0
            right = len(curArray) - 1
            while left <= right:
                curMiddle = (left + right) // 2
                if curArray[curMiddle] == target:
                    return True
                elif curArray[curMiddle] < target:
                    left = curMiddle + 1
                elif curArray[curMiddle] > target:
                    right = curMiddle - 1
            return False
        if len(matrix) == 0:
            return False
        elif len(matrix) == 1:
            return binarySearchRow(matrix[0])
        else:
            i = 0
            while i < len(matrix) - 1:
                if binarySearchRow(matrix[i] + matrix[i+1]):
                    return True
                i += 2
            if len(matrix) % 2 != 0:
                return binarySearchRow(matrix[i])
            return False

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-4.cpp
class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        if (nums.size() == 1) {
            if (nums[0] == target) {
                return 1;
            }
            return 0;
        }
        int left = 0;
        int right = 0;
        int minLength = nums.size() + 1;
        int curSum = nums[left];
        while (right < nums.size()) {
            //printf("Left Value: %d\n", left);
            //printf("Right Value: %d\n", right);
            if  (curSum >= target) {
                //printf("Found length: %d\n", right - left + 1);
                //printf("Current min length: %d\n", minLength);
                if (right - left + 1 < minLength) {
                    minLength = right - left + 1;
                    //printf("New min length: %d\n", minLength);
                }
                curSum -= nums[left];
                left += 1;
            }
            else {
                right += 1;
                if (right < nums.size()) {  
                    curSum += nums[right];
                }
            }
        } 
        if (minLength == nums.size() + 1) {
            return 0;
        }
        return minLength;
    }
};